apollo-router/src/configuration/cors.rs

```diff
@@ -254,88 +338,96 @@ impl Cors {
     //
     // Similarly, `Access-Control-Expose-Headers`, `Access-Control-Allow-Methods`, and `Access-Control-Allow-Headers`
     // response headers can only use `*` as value when request's credentials mode is not "include".
-    pub(crate) fn ensure_usable_cors_rules(&self) -> Result<(), &'static str> {
+    pub(crate) fn ensure_usable_cors_rules(&self) -> Result<(), CorsConfigError> {
         // Check for wildcard origins in any Policy
         if let Some(policies) = &self.policies {
-            for policy in policies {
-                if policy.origins.iter().any(|x| x == "*") {
-                    return Err(
-                        "Invalid CORS configuration: use `allow_any_origin: true` to set `Access-Control-Allow-Origin: *`",
-                    );
+            for policy in policies.iter() {
+                if policy.origins.iter().any(|x| &**x == "*") {
```

Comment:
My understanding is that for anything that is contained within a smart pointer type `as_ref` is more idiomatic than `&**` or variants of.

Reply:
Good point. Addressed in 0bf34ba08